src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,14 +6,12 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact{
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
-    private boolean isPrivate;
 
```

Comment:
You can abstract it even more by having a constructor within `Contact` class. By passing in `MESSAGE_ADDRESS_CONSTRAINTS` in the super() constructor call, you can even perform validation in the super class.